var/spack/repos/builtin/packages/improved-rdock/package.py

```diff
@@ -61,12 +61,22 @@ def install(self, spec, prefix):
     def setup_run_environment(self, env):
         env.set("RBT_ROOT", self.prefix)
 
-    def test(self):
+    def run_rdock_test(self, exe, opts):
+        """Easy reuse test method"""
         copy(join_path(self.prefix.example, "1sj0", "*"), ".")
-        opts = ["-r", "1sj0_rdock.prm", "-was"]
-        self.run_test("rbcavity", options=opts)
+        exe = which(exe)
+        exe(*opts)
 
-        mpiexe = self.spec["mpi"].prefix.bin.mpirun
+    def test_rbcavity(self):
+        """Check rbcavity"""
+        self.run_rdock_test("rbcavity", ["-r", "1sj0_rdock.prm", "-was"])
+
+    def test_bash(self):
```

Comment:
This will not be executed in the correct order if it is pulled out as a separate (implicit) test part given that it produces